### PR TITLE
Code generator Win support

### LINKED ISSUE #4: CI fails on Win
Right now `go test ./internal/code-generator/... -run Test_GeneratorTestSuite -testify.m TestGenerationFiles` fails on CI on master only for the Windows platform, most probably due to some path-related issues.

### PR DESCRIPTION
**Pull request by bzz**
Fixes #4 

On Win `make code-generate` produces unreasonable Bayesian classifier weights from Linguist samples silently, failing only the final classification tests.

TestPlan:
 - passing tests on Win CI 
   ```
   go test ./internal/code-generator/... \
    -run Test_GeneratorTestSuite -testify.m TestGenerationFiles
   ```

### CODE REVIEW

File: internal/code-generator/main.go [hunk 1]
```diff
@@ -125,7 +128,7 @@ func getCommit(path string) (string, error) {
 	}
 
 	if string(commit) == "ref: refs/heads/master\n" {
-		path = ".linguist/.git/" + string(commit[5:len(commit)-1])
+		path = filepath.Join(".linguist", ".git/", string(commit[5:len(commit)-1]))
```
Comment (lafriks) [inline internal/code-generator/main.go]:
```suggestion
		path = filepath.Join(".linguist", ".git", string(commit[5:len(commit)-1]))
```

**Comment by bzz** [comment]
Reading samples for Bayesian classifier

### Before 
*nix: `Total samples: 2187`
Win: `Total samples: 2189`

**Comment by lafriks** [comment]
Still fails only on Windows :]

**Comment by bzz** [comment]
> Still fails only on Windows :]

yup, a one more, different bug 🤦‍♂

**Comment by bzz** [comment]
It seems there is a different number of tokens at least for the [ATS language samples](https://github.com/github/linguist/tree/master/samples/ATS) on windows
Screen Shot 2020-03-28 at 10 14 40 PM

**Comment by bzz** [comment]
Results of the same tokenizer test on the same file:
```
 go test ./internal/code-generator/generator -run Test_GeneratorTestSuite -testify.m TestTokenizerOnATS
```
 
win
```
generator_test.go:324: File: "C:\\Users\\RUNNER~1\\AppData\\Local\\Temp\\linguist-512056635\\samples\\ATS\\csv_parse.hats", tokens: 463
```

mac
```
generator_test.go:324: File: "go-enry/.linguist/samples/ATS/csv_parse.hats", tokens: 381
```

**Comment by bzz** [comment]
[Tokenizer regex](https://github.com/go-enry/go-enry/blob/master/internal/tokenizer/tokenize.go#L81) rely on matching platform-specific EOL 🤕 that causes the number of tokens used for building bayesian classifier to be different using the same samples on different platforms.

**Comment by bzz** [comment]
The number of tokes look the same now, but the test is still failing because the order of keys in 2 maps (original, generated on *nix and one on win CI) is different: even if the values are the same we are using string comparison 🔫

**Comment by bzz** [comment]
Merging as CI is green now.

**Comment by lafriks** [comment]
Great work!

**Comment by bzz** [comment]
For the history 1724869 applies https://github.com/actions/checkout/issues/135 as a workaround to keep the CI green bespite [platform-dependent regex in default Tokenizer]( https://github.com/go-enry/go-enry/blob/master/internal/tokenizer/tokenize.go#L81).

**Pull request merged**